src/SchemaDesign/SchemaDesignerQuestion.tsx

```diff
@@ -0,0 +1,81 @@
+/* eslint-disable dot-notation */
+import { SchemaQuestion } from '@/types/interfaces'
+import Box from '@mui/material/Box'
+import Button from '@mui/material/Button'
+import Stack from '@mui/material/Stack'
+import TextField from '@mui/material/TextField'
+import React from 'react'
+
+export default function SchemaDesignerQuestion({
+  onSubmit,
+  question,
+  closeQuestionPicker,
+}: {
+  onSubmit: (data: SchemaQuestion) => void
+  question: any // type this
+  closeQuestionPicker: () => void
+}) {
+  const [formInput, setFormInput] = React.useState<SchemaQuestion>({
+    title: '',
+    description: '',
+    type: question.type,
+    reference: '',
+    format: question.format || '',
+  })
+
+  const handleChange = (event: React.ChangeEvent<HTMLTextAreaElement | HTMLInputElement>) => {
+    const { value, name }: { value: string | number; name: string } = event.target
+    if (value || (!value && event.target.attributes['required'])) {
+      setFormInput({
+        ...formInput,
+        [event.target.name]: value,
+      })
+    } else {
+      setFormInput((prevData) => {
+        const newData = { ...prevData }
+        delete newData[name]
+        return newData
+      })
+    }
+  }
+
+  const addQuestion = (event: React.FormEvent<HTMLInputElement>) => {
+    event.preventDefault()
+    onSubmit(formInput)
+    closeQuestionPicker()
+  }
+
+  return (
+    <Box component='form' onSubmit={addQuestion}>
+      <Stack spacing={2}>
+        <TextField label='Question reference' name='reference' value={formInput.reference} onChange={handleChange} />
+        <TextField required label='Question text' name='title' value={formInput.title} onChange={handleChange} />
+        <TextField
+          label='Question description'
+          name='description'
+          value={formInput.description}
+          onChange={handleChange}
+        />
+        {question.additionalQuestions.includes('minLength') && (
+          <TextField
+            label='Minimum length'
+            type='number'
```

Comment:
`number` inputs are very old and do not conform to modern accessibility standards for what we're using it for here. Instead you should use `type="text" inputmode="numeric" pattern="[0-9]*"` for the most comprehensive number input field.

If we want to limit the input to only accept certain characters, it's much better if we use form validation instead of blocking values outside of a given set of values. For example, if you use `type='number'` and try to put in alphabetic characters nothing will appear in the input field, but the user isn't given any feedback as to why their keystrokes have been ignored. We can instead validate the value when they try to submit the form and provide a useful error/helper text message telling them what values the field accepts.

You can easily fall into a bit of a rabbit hole when looking into this so here's a few useful sources that help to describe the problem:
https://developer.mozilla.org/en-US/docs/Web/HTML/Element/input/number#using_number_inputs
https://developer.mozilla.org/en-US/docs/Web/HTML/Element/input/number#accessibility

Reply:
According to the MUI docs, <TextArea ..> should use "type: number". Looking at the generated HTML, it creates an <input /> element with "type: number". Not sure how best to approach this - should we use MUI TextArea, or should we use base <input />? The attribute of "pattern" isn't valid for MUI's TextArea

Reply:
You're using `TextField` here, not `TextArea`.

Could you link me to the documentation you're referring to? I can see that MUI provides a couple of `TextArea`s via the `@mui/joy` and `@mui/base` packages, but we use `@mui/material` which doesn't offer a `TextArea`, so a bit confused as to what you're referring to.

Reply:
apologies I meant <TextField /> :)

Reply:
There's still a bug with the minLengh and maxLength fields so for now I might disable them anyway, as I want to draw a line in the scope of this ticket.